addons/mail/models/mail_thread.py

```diff
@@ -2037,8 +2037,9 @@ def _message_set_main_attachment_id(self, attachment_ids):
             ])
             prioritary_attachments = all_attachments.filtered(lambda x: x.mimetype.endswith('pdf')) \
                                      or all_attachments.filtered(lambda x: x.mimetype.startswith('image')) \
-                                     or all_attachments
-            self.with_context(tracking_disable=True).write({'message_main_attachment_id': prioritary_attachments[0].id})
+                                     or all_attachments.filtered(lambda x: not x.mimetype.endswith('xml'))
+            if prioritary_attachments:
+                self.sudo().with_context(tracking_disable=True).write({'message_main_attachment_id': prioritary_attachments[0].id})
```

Comment:
Why do you add a sudo here ?

Reply:
A ghost from previous versions. He's gone now. :ghost: :no_entry_sign: